docs/apispec-docs/src/main/scala/sttp/tapir/docs/apispec/schema/TSchemaToASchema.scala

```diff
@@ -26,7 +26,12 @@ private[schema] class TSchemaToASchema(toSchemaReference: ToSchemaReference, mar
       case TSchemaType.SArray(nested @ TSchema(_, Some(name), _, _, _, _, _, _, _, _, _)) =>
         ASchema(SchemaType.Array).copy(items = Some(toSchemaReference.map(nested, name)))
       case TSchemaType.SArray(el) => ASchema(SchemaType.Array).copy(items = Some(apply(el)))
-      case TSchemaType.SOption(nested @ TSchema(_, Some(name), _, _, _, _, _, _, _, _, _)) => toSchemaReference.map(nested, name)
+      case TSchemaType.SOption(nested @ TSchema(_, Some(name), _, _, _, _, _, _, _, _, _)) => {
+        val ref = toSchemaReference.map(nested, name)
+        if (!markOptionsAsNullable) ref
+        else
+          ASchema.oneOf(List(ref), ref.discriminator).copy(nullable = Some(true))
```

Comment:
maybe doing here simply `ASchema.oneOf(List(ref, ASchema(SchemaType.Null), None)` would work?

Reply:
I've added this change, let's see if things work :)

Reply:
great! thank you :) 
what else is needed to merge the PR?

Reply:
nothing, tests pass :)

Reply:
thanks!